src/cloudWatchLogs/cloudWatchLogsUtils.ts

```diff
@@ -6,28 +6,57 @@
 import * as vscode from 'vscode'
 import { CLOUDWATCH_LOGS_SCHEME } from '../shared/constants'
 import { fromExtensionManifest } from '../shared/settings'
+import { CloudWatchLogsGroupInfo, parametersStringValue } from './registry/logStreamRegistry'
+import { CloudWatchLogsParameters } from './registry/logStreamRegistry'
 
 // URIs are the only vehicle for delivering information to a TextDocumentContentProvider.
 // The following functions are used to structure and destructure relevant information to/from a URI.
 // Colons are not valid characters in either the group name or stream name and will be used as separators.
 
+let parsingURIMap = new Map<string, URIParsingFunction>()
+
+parsingURIMap.set('viewLogStream', function (parts) {
```

Comment:
Might not need a `parsingURIMap` datastructure. Could start with a switch-case in `parseCloudWatchLogsUri()`. Then see if the need for something more abstract becomes very obvious.